Components/Pages/Goals.js

```diff
@@ -87,40 +78,27 @@ const Goals = ({ navigation }) => {
     }
   };
 
-  const _handleNotification = notification => {
-    Vibration.vibrate();
-    setNotification({ notification: notification });
-  };
-
-  const attachToken = async () => {
-    AsyncStorage.getItem("notification_token").then(token => {
-      setnotificationsEnabled(token !== null);
-    });
+  const attachToken = async expoPushToken => {
     const apiToken = await AsyncStorage.getItem("token");
-
-    if (!notificationsEnabled) {
-      if (expoPushToken) {
-        await fetch(NOTIFICATION_URI, {
-          method: "POST",
-          headers: {
-            "Accept": "application/json",
-            "Accept-encoding": "gzip, deflate",
-            "Content-Type": "application/json",
-            "Authorization": `Bearer ${apiToken}`,
-          },
-          body: JSON.stringify({ expoPushToken: expoPushToken }),
-        }).then(() => {
-          AsyncStorage.setItem("notification_token", expoPushToken);
```

Comment:
since you set the notification_token to async after its sent to server, you could get the token on line 100 with the below. If it returns null, then send the fetch request. If it does not return null, then we know that the token was already sent and it doesnt need to be sent again

Reply:
@ChrisCruze yeah that makes sense. Why send it again if its been done already, right? I was thinking about something like that with this comment: https://github.com/ChrisCruze/Push-IOS/pull/50/files#diff-04508f5b5d8253b118caf3d90e949c46R95

Let me add it in. Thanks for good feedback!